tests: Add comprehensive unit tests for OrderService

JzomitiZumatti opened the pull request:
This commit adds unit tests to ensure the correct behavior of methods in the OrderService class. The tests cover various scenarios including creating an order, retrieving orders by ID, updating orders, adding products to orders, deleting products from orders, and deleting orders by ID. Each test validates the expected behavior of the corresponding method under different conditions, ensuring proper handling of edge cases and error scenarios. Additionally, Mockito is used to mock dependencies such as the OrderRepository, allowing isolated testing of the OrderService functionality. These tests provide comprehensive coverage to ensure the reliability and correctness of the OrderService implementation.

JzomitiZumatti committed:
tests: Add comprehensive unit tests for OrderService

This commit adds unit tests to ensure the correct behavior of methods in the OrderService class. The tests cover various scenarios including creating an order, retrieving orders by ID, updating orders, adding products to orders, deleting products from orders, and deleting orders by ID. Each test validates the expected behavior of the corresponding method under different conditions, ensuring proper handling of edge cases and error scenarios. Additionally, Mockito is used to mock dependencies such as the OrderRepository, allowing isolated testing of the OrderService functionality. These tests provide comprehensive coverage to ensure the reliability and correctness of the OrderService implementation.

IvashchenkoIvanIgorevich commented on src/test/java/com/example/glovodb/service/OrderServiceTest.java:
```diff
@@ -1,5 +1,159 @@
+
+    @Test
+    public void addProductToOrderTest() {
+        Collection<Product> products = new ArrayList<>();
+        products.add(new Product());
```
You can simplify that by using `Collections.singletonList(new Product());` and past it to the products fields in builder

JzomitiZumatti replied:
I tried but got "java.lang.UnsupportedOperationException", because than I'm trying to add product to immutable collection.

IvashchenkoIvanIgorevich replied:
Yes, my bad, sorry.

IvashchenkoIvanIgorevich commented on src/test/java/com/example/glovodb/service/OrderServiceTest.java:
Please make refactoring by adding empty lines between Arrange, Act, and Assert.

IvashchenkoIvanIgorevich reviewed (approved):
Good work!

JzomitiZumatti committed:
refactor: Update OrderServiceTest class

Updated OrderServiceTest class to improve readability and maintainability. Removed unnecessary imports and streamlined test methods for better clarity. Introduced consistent naming conventions for test methods and variables. Utilized Mockito for mocking dependencies, ensuring isolated testing of OrderService functionality. Reviewed and aligned code structure with best practices, enhancing overall code quality and maintainability.
